### PR TITLE
feat/1903-relatedOptions

### LINKED ISSUE #1903: Add relatedOptions array to config options
Optional

Each config option in lib/config/definitions.js should have an optional field “relatedOptions” which lists other options that are similar or related. This will be used later in autogenerate docs. 

Should ideally also have a linting to make sure there’s no typos or deprecated options linked, i.e. make sure that every entry in every `relatedOptions` field exists.

### PR DESCRIPTION
**Pull request by dragomirt**
This PR adds a linter for the optional `relatedOptions` field in definitions.js which checks if the referenced options exist.

Closes #1903

### CODE REVIEW

File: test/website-docs.spec.js [hunk 1]
```diff
@@ -51,4 +51,18 @@ describe('docs', () => {
...
+    .filter(option => option.relatedOptions)
+    .map(option => option.relatedOptions)
+    .sort();
+
+  relatedOptions = [].concat(...relatedOptions);
+
+  const allOptionNames = options.map(option => option.name).sort();
+
+  it('has valid relateOptions values', () => {
+    expect(allOptionNames).toEqual(expect.arrayContaining(relatedOptions));
```
Comment (rarkins) [inline test/website-docs.spec.js]: I'm not familiar with this syntax. Is this the same as "expect relatedOptions to be a subset of allOptionNames" ?

File: test/website-docs.spec.js [hunk 2]
```diff
@@ -51,4 +51,18 @@ describe('docs', () => {
...
     expect(headers3).toEqual(expectedOptions3);
   });
+
+  // Checking relatedOptions field in definitions
+  let relatedOptions = options
+    .filter(option => option.relatedOptions)
+    .map(option => option.relatedOptions)
+    .sort();
+
+  relatedOptions = [].concat(...relatedOptions);
```
Comment (rarkins) [inline test/website-docs.spec.js]: What does this achieve?

File: test/website-docs.spec.js (hunk 1 above)
Comment (rarkins) [reply test/website-docs.spec.js]: Is the error message meaningful to the user if they make a mistake? e.g. prints the incorrect option? Maybe a for loop and explicit error message would be better

**Review by rarkins — CHANGES_REQUESTED** [review]
Improve error printing

File: test/website-docs.spec.js (hunk 2 above)
Comment (SevenTurtles) [reply test/website-docs.spec.js]: The last line converts the 2d matrix to 1d to have a valid subset for search. E.g. `[['enabled', 'packageRules']]` => `['eanbled', 'packageRules']`

**Pull request merged**